internal/framework/provider/authenticationv1/ephemeral_token_request_v1.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) HashiCorp, Inc.
+
+package authenticationv1
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-provider-kubernetes/kubernetes"
+
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	_ ephemeral.EphemeralResource              = (*TokenRequestEphemeralResource)(nil)
+	_ ephemeral.EphemeralResourceWithConfigure = (*TokenRequestEphemeralResource)(nil)
+)
+
+type TokenRequestEphemeralResource struct {
+	SDKv2Meta func() any
+}
+
+type TokenRequestMetadata struct {
+	Name      types.String `tfsdk:"name"`
+	Namespace types.String `tfsdk:"namespace"`
+}
+
+type BoundObjectReference struct {
+	APIVersion types.String `tfsdk:"api_version"`
+	Kind       types.String `tfsdk:"kind"`
+	Name       types.String `tfsdk:"name"`
+	UID        types.String `tfsdk:"uid"`
+}
+
+type TokenRequestSpec struct {
+	Audiences         []types.String        `tfsdk:"audiences"`
+	BoundObjecRef     *BoundObjectReference `tfsdk:"bound_object_ref"`
+	ExpirationSeconds types.Int64           `tfsdk:"expiration_seconds"`
+}
+
+type TokenRequestModel struct {
+	Metadata TokenRequestMetadata `tfsdk:"metadata"`
+	Spec     *TokenRequestSpec    `tfsdk:"spec"`
+
+	Token               types.String `tfsdk:"token"`
+	ExpirationTimestamp types.String `tfsdk:"expiration_timestamp"`
+}
+
+func NewTokenRequestEphemeralResource() ephemeral.EphemeralResource {
+	return &TokenRequestEphemeralResource{}
+}
+
+func (r *TokenRequestEphemeralResource) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.SDKv2Meta = req.ProviderData.(func() any)
```

Comment:
I have never done this before

Reply:
This is to avoid fully duplicating all of the client initialization code and potentially introducing more inconsistencies between SDKv2 and Framework. When all of the SDKv2 resources are moved to Framework the initialization code can be ported as the last step.